sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/policy/ProviderRegistrationPolicy.java

```diff
@@ -78,7 +78,7 @@ public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineN
         if (providers == null) {
             return next.process();
         }
-        return next.process().flatMap(
+        return next.clone().process().flatMap(
```

Comment:
May I know what's the use of added `clone()` here?

Reply:
Cannot say I am exactly sure. It is done in RetryPolicy https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

I think it run the request on the clone of the "next pipeline", so that if one need to run it again, the current "next pipeline" is not affected by the previous run of cloned one.